<a href="https://colab.research.google.com/github/vaishnavi17398/FeynnLabsInternship/blob/main/mental_health.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
uploaded = files.upload()


Saving Mental health Depression disorder Data.csv to Mental health Depression disorder Data.csv


In [2]:
import pandas as pd
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt

In [10]:
data = pd.read_csv("Mental health Depression disorder Data.csv",error_bad_lines=False)

<ipython-input-10-a31b09f117a8>:1: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  data = pd.read_csv("Mental health Depression disorder Data.csv",error_bad_lines=False)
<ipython-input-10-a31b09f117a8>:1: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv("Mental health Depression disorder Data.csv",error_bad_lines=False)


In [11]:
# Drop non-relevant columns
data = data.drop(columns=['index', 'Entity', 'Code', 'Year'])

In [13]:
# Exclude non-numeric columns (assuming they are not relevant for clustering)
numeric_data = data.select_dtypes(include=['float64', 'int64'])

In [15]:
# Normalize the data
scaler = StandardScaler()
scaled_data = scaler.fit_transform(numeric_data)

In [7]:
# Choose the number of clusters (you can use elbow method or silhouette score)
num_clusters = 3

In [23]:
# Apply K-means clustering
# Apply K-means clustering
kmeans = KMeans(n_clusters=3, random_state=42)
kmeans.fit(scaled_data)
labels = kmeans.labels_

/usr/local/lib/python3.10/dist-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(


In [19]:
# Impute missing values with the mean of each column
from sklearn.impute import SimpleImputer
imputer = SimpleImputer(strategy='mean')
imputed_data = pd.DataFrame(imputer.fit_transform(numeric_data), columns=numeric_data.columns)


In [21]:
# Scale the imputed numeric data
scaler = StandardScaler()
scaled_data = scaler.fit_transform(imputed_data)

In [24]:
# Add cluster labels to the original dataframe
data['Cluster'] = labels

In [26]:
# Analyze results
cluster_centers = scaler.inverse_transform(kmeans.cluster_centers_)
cluster_df = pd.DataFrame(cluster_centers, columns=numeric_data.columns)  # Use numeric_data.columns

In [27]:
# Print cluster centers
print("Cluster Centers:")
print(cluster_df)

Cluster Centers:
   Anxiety disorders (%)  Drug use disorders (%)  Depression (%)  \
0               3.972510                0.853771        3.488979   
1               5.601623                1.541565        4.069338   
2               2.909291                0.744885        3.837485   

   Alcohol use disorders (%)  
0                   1.580840  
1                   1.276212  
2                   4.202753  


Cluster 0:


Anxiety disorders (%): The prevalence of anxiety disorders in this cluster is moderate, with a mean value of approximately 3.97%.
Drug use disorders (%): Drug use disorders are relatively low in this cluster, with a mean value of around 0.85%.
Depression (%): The prevalence of depression is moderate, with a mean value of approximately 3.49%.
Alcohol use disorders (%): Alcohol use disorders are relatively high in this cluster, with a mean value of about 1.58%.

Cluster 1:



Anxiety disorders (%): This cluster exhibits a higher prevalence of anxiety disorders compared to Cluster 0, with a mean value of approximately 5.60%.
Drug use disorders (%): Drug use disorders are also higher in this cluster compared to Cluster 0, with a mean value of around 1.54%.
Depression (%): The prevalence of depression is similar to Cluster 0, with a mean value of approximately 4.07%.
Alcohol use disorders (%): Alcohol use disorders are lower in this cluster compared to Cluster 0, with a mean value of about 1.28%.


Cluster 2:
Anxiety disorders (%): The prevalence of anxiety disorders in this cluster is lower compared to both Cluster 0 and Cluster 1, with a mean value of approximately 2.91%.
Drug use disorders (%): Drug use disorders are relatively low in this cluster, with a mean value of around 0.74%.
Depression (%): Interestingly, the prevalence of depression is higher in this cluster compared to Cluster 0, with a mean value of approximately 3.84%.
Alcohol use disorders (%): Alcohol use disorders are significantly higher in this cluster compared to both Cluster 0 and Cluster 1, with a mean value of about 4.20%.


Conclusion:
Cluster 1 appears to represent a group with higher rates of anxiety disorders and drug use disorders compared to Clusters 0 and 2.
Cluster 0 shows moderate rates of anxiety disorders, depression, and higher rates of alcohol use disorders.
Cluster 2 represents a group with lower rates of anxiety disorders but higher rates of depression and alcohol use disorders compared to the other clusters.